tests.py

```diff
@@ -1,5 +1,12 @@
+from copy import deepcopy
```

Comment:
Please, move this `from ... import` below the pure `import` statements. Alphabetic order, preferably.